tests/components/watergate/test_valve.py

```diff
@@ -0,0 +1,45 @@
+"""Tests for the Watergate coordinator."""
+
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.valve import DOMAIN as VALVE_DOMAIN, ValveState
+from homeassistant.const import ATTR_ENTITY_ID, SERVICE_CLOSE_VALVE, SERVICE_OPEN_VALVE
+from homeassistant.core import HomeAssistant
+
+from . import init_integration
+
+
+async def test_change_valve_state(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_watergate_client,
+    mock_entry,
+) -> None:
+    """Test entities become unavailable after failed update."""
+    await init_integration(hass, mock_entry)
+    entity_id = "valve.sonic"
+
+    registered_entity = hass.states.get(entity_id)
+    assert registered_entity
+    assert registered_entity.state == ValveState.OPEN
+    assert registered_entity == snapshot
+
+    await hass.services.async_call(
+        VALVE_DOMAIN,
+        SERVICE_CLOSE_VALVE,
+        {ATTR_ENTITY_ID: entity_id},
+        blocking=True,
+    )
+
+    assert hass.states.get(entity_id).state == ValveState.CLOSING
+
+    mock_watergate_client.async_set_valve_state.assert_called_once_with("close")
+
+    await hass.services.async_call(
+        VALVE_DOMAIN,
+        SERVICE_OPEN_VALVE,
+        {ATTR_ENTITY_ID: entity_id},
+        blocking=True,
+    )
+
+    assert hass.states.get(entity_id).state == ValveState.OPENING
```

Comment:
```suggestion
    registered_entity = hass.states.get(entity_id)
    assert registered_entity
    assert registered_entity.state == ValveState.OPENING
```
that's what I meant with "assert state before". I'm surprised that mypy doesn't complain otherwise...